### PR TITLE
Traffic detectors weekly snapshot DAG

### LINKED ISSUE #19844: Would like to create additional Traffic Detector dataset using existing datafeed
> What application are you using?

AMD Data Tracker

> Describe the problem.

Would like to create additional dataset using existing datafeed: https://datahub.austintexas.gov/Transportation-and-Mobility/Traffic-Detectors/qpuw-8eeb/about_data

-inserts new rows weekly by detector 
-internal datahub only (no need for external)
-data retention: 5 years

Go by: [Ronnie inventory transactions dataset](https://datahub.austintexas.gov/Transportation-and-Mobility/Arterial-Management-Materials-Warehouse-Inventory/hcaw-evi2/about_data)

> Describe the outcome you'd like to see when this feature is implemented.

Would like to create additional dataset using existing datafeed: https://datahub.austintexas.gov/Transportation-and-Mobility/Traffic-Detectors/qpuw-8eeb/about_data

-inserts new rows weekly by detector 
-internal datahub only (no need for external)
-data retention: 5 years

Go by: Ronnie inventory transactions dataset

> Describe any workarounds you currently have in place or alternative solutions you've considered.

none - no way to historically archive outside of manual export and save

> Is there anything else we should know?

End of November

> Requested By
Allyson R.

Request ID: DTS24-119080

### PR DESCRIPTION
**Pull request by Charlie-Henry**
## Associated issues
closes https://github.com/cityofaustin/atd-data-tech/issues/19844

## Associated repo
https://github.com/cityofaustin/atd-knack-services

## Testing

**Steps to test:**
```
docker compose run --rm airflow-cli dags test atd_knack_traffic_detectors_weekly_snapshot
```
**or**

`docker compose up` -> trigger `atd_knack_traffic_detectors_weekly_snapshot` through the UI

Then:

- Check the [dataset](https://datahub.austintexas.gov/Transportation-and-Mobility/Traffic-Detectors-Weekly-Archive/g939-dyvz/about_data), to see if it grows by ~5k records and the `published_date` column should have data from when you triggered it.
- Check the socrata backups [S3 bucket for this dataset](https://us-east-1.console.aws.amazon.com/s3/buckets/atd-socrata-backups?region=us-east-1&bucketType=general&prefix=g939_dyvz/) for a backup csv file for the date that you ran the script.

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Add note to 1PW secrets moved to API vault and check for duplicates

### CODE REVIEW

File: dags/atd_knack_traffic_detectors_weekly_snapshot.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        "opfield": "production.Bucket",
+    },
+}
+
+
+with DAG(
+    dag_id="atd_knack_traffic_detectors_weekly_snapshot",
+    description="Appends traffic detector assets to a running log in Socrata",
+    default_args=DEFAULT_ARGS,
+    schedule_interval="0 1 * * SUN" if DEPLOYMENT_ENVIRONMENT == "production" else None,
```
Comment (Charlie-Henry) [inline dags/atd_knack_traffic_detectors_weekly_snapshot.py]: Runs 1AM on Sundays
[0 1 * * SUN](https://crontab.guru/#0_1_*_*_SUN)

Comment (Charlie-Henry) [reply dags/atd_knack_traffic_detectors_weekly_snapshot.py]: @ChristinaTremel is this a good time?

Comment (Charlie-Henry) [reply dags/atd_knack_traffic_detectors_weekly_snapshot.py]: Hmm now that I think about it, maybe we should not use the weekend as that's usually when socrata is down for maintenance. Would Monday work? I guess there's no problem running this daily at 1AM either. I guess I'm worried about missing a week every now and again.

**Review by chiaberry — APPROVED** [review]
ran locally, got all green squares and saw my updates on socrata 👍🏼

File: dags/atd_knack_traffic_detectors_weekly_snapshot.py (hunk 1 above)
Comment (johnclary) [reply dags/atd_knack_traffic_detectors_weekly_snapshot.py]: yes, i think avoiding the weekends is a good call. do you want to add retries? i see you have it set to `0` at the moment.

**Review by johnclary — APPROVED** [review]
Looks good 🚢

**Review by mddilley — APPROVED** [review]
Ran this successfully, and I see the new rows in the dataset and a new .csv in the S3 bucket. ✅🚢

File: dags/atd_knack_traffic_detectors_weekly_snapshot.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# test locally with: docker compose run --rm airflow-cli dags test atd_knack_traffic_detectors_weekly_snapshot
```
Comment (frankhereford) [inline dags/atd_knack_traffic_detectors_weekly_snapshot.py:1]: thanks for these helpful testing breadcrumbs, and doubly so for doing them with the compose stack -- that's 💯

File: dags/atd_knack_traffic_detectors_weekly_snapshot.py (hunk 1 above)
Comment (frankhereford) [reply dags/atd_knack_traffic_detectors_weekly_snapshot.py]: > Uppercase, and lowercase or mixed-case three character strings, based on the English name of the day: mon, tue, wed, thu, fri, sat, or sun.

Never in 25 years have I seen this, and it's widely supported. Works for months too, `Jan`, `Oct`, `Dec`, etc. I have struggled keeping the numbers of months straight for my entire life, and this end-runs this. TIL of the year.

**Review by frankhereford — APPROVED** [review]
Tested - works great. Orchestration here is on point, and the ETL appears to have done what it should too. Thanks for the mega-TIL on cron expressions. 🙏

File: dags/atd_knack_traffic_detectors_weekly_snapshot.py (hunk 1 above)
Comment (Charlie-Henry) [reply dags/atd_knack_traffic_detectors_weekly_snapshot.py]: sounds good, moved it to 1AM Monday. And added retries to hopefully have less gaps in data!

**Pull request merged**